code/obj/sawflymisc.dm

```diff
@@ -0,0 +1,630 @@
+		beeptext = "[pick(list("beeps",  "boops", "bwoops", "bips", "bwips", "bops", "chirps", "whirrs", "pings", "purrs", "thrums"))]"
+		animate_bumble(src) // gotta get the float goin' on
+		src.set_a_intent(INTENT_HARM) // incredibly stupid way of ensuring they aren't passable
+
+
+
+	proc/foldself()
+		var/obj/item/old_grenade/sawfly/reused/N = new /obj/item/old_grenade/sawfly/reused(get_turf(src))
+		// pass our name and health
+		N.name = "Compact [name]"
+		N.tempname = src.name
+		N.temphp = src.health
+		qdel(src)
+
+	proc/communalbeep() // distribues the beepchance among the number of sawflies nearby
+		fliesnearby = 1 //that's you, little man! :)
+		for(var/mob/living/critter/sawfly/E in range(get_turf(src), 18))
+			src.fliesnearby += 1 //that's your buddies!
+		var/beepchance = (1 / fliesnearby) * 100 //if two sawflies, give 50% chance that any one will beep
+		if(fliesnearby<3) beepchance -=20 //heavily reduce chance of beep in swarm
+		if(prob(beepchance))
+			playsound(src, pick(src.beeps), 40, 1)
+			src.visible_message("<b>[src] [beeptext].</b>")
+
+
+
+	emp_act() //same thing as if you emagged the controller, but much higher chance
+		if(prob(80))
+			src.visible_message("<span class='combat'>[src] buzzes oddly and starts to sprial out of contro!</span>")
+			SPAWN(2 SECONDS)
+				src.blowup()
+		else
+			src.foldself()
+
+
+	Cross(atom/movable/mover) //code that ensure projectiles hit them when they're alive, but won't when they're dead
+		if(istype(mover, /obj/projectile))
+			if(!src.alive)
+				return 1
+		return ..()
+
+
+	death()
+		CritterDeath()
+
+	proc/CritterDeath() //  SUPER important proc do NOT touch this or everything will break and You Will Cry
+
+		if (alreadydead) return // prevents any death behavior from ever happening more than once
+
+		alive = FALSE
+		src.set_density(0)
+		src.force_laydown_standup()
+		src.tokenized_message(death_text)
+		src.is_npc = FALSE // stop any and all possible non-critter AI thought
+		src.throws_can_hit_me = FALSE  //prevent getting hit by thrown stuff- super important in avoiding jank
+
+
+		animate(src) //stop no more float animation
+		icon_state = "sawflydead[pick("1", "2", "3", "4", "5", "6", "7", "8")]" //randomly selects death icon and displaces them
+		src.pixel_x += rand(-5, 5)
+		src.pixel_y += rand(-1, 5)
+		src.anchored = 0
+		walk_to(src,0) //halt walking
+
+		// special checks that determine how much damage they do after death
+		if (prob(20))
+			new /obj/item/device/prox_sensor(src.loc)
+			return
+
+		if(prob(60))
+			elecflash(src, 1, 3)
+
+		if(prob(20)) // congrats, little guy! You're special! You're going to blow up!
+			if(prob(70)) //decide whether or not people get a warning
+				src.visible_message("<span class='combat'>[src] makes a [pick("gentle", "odd", "slight", "weird", "barely audible", "concerning", "quiet")] [pick("hiss", "drone", "whir", "thump", "grinding sound", "creak", "buzz", "khunk")].......")
```

Comment:
```suggestion
				src.visible_message("<span class='combat'>[src] makes \a [pick("gentle", "odd", "slight", "weird", "barely audible", "concerning", "quiet")] [pick("hiss", "drone", "whir", "thump", "grinding sound", "creak", "buzz", "khunk")].......")
```
Not sure if text macro works like this, I don't have much experience with them. But the adjectives sometimes necessitate an "an".

Also, the amount of dots seems a bit excessive......